test/config-flags-gulpfile.js

```diff
@@ -1,129 +1,135 @@
 'use strict';
 
 var expect = require('expect');
-
+var exec = require('child_process').exec;
 var path = require('path');
-var fixturesDir = path.join(__dirname, 'fixtures/config');
 
-var headLines = require('gulp-test-tools').headLines;
-var eraseTime = require('gulp-test-tools').eraseTime;
-var runner = require('gulp-test-tools').gulpRunner().basedir(fixturesDir);
+var sliceLines = require('./tool/slice-lines');
+var cmdSep = require('./tool/cmd-sep');
+
+var gulpCmd = 'node ' + path.join(__dirname, '../bin/gulp.js');
+var baseDir = path.join(__dirname, 'fixtures/config/flags/gulpfile');
 
 describe('config: flags.gulpfile', function() {
 
   it('Should configure with a .gulp.* file', function(done) {
-    runner
-      .chdir('flags/gulpfile')
-      .gulp()
-      .run(cb);
+    exec([
+      'cd ' + baseDir + cmdSep,
+      gulpCmd,
+    ].join(' '), cb);
 
     function cb(err, stdout, stderr) {
-      expect(err).toEqual(null);
+      expect(err).toBeNull();
       expect(stderr).toEqual('');
-      stdout = headLines(stdout, 2, 2);
-      expect(stdout).toEqual(
-        'This gulpfile : ' +
-          path.join(fixturesDir, 'flags/gulpfile/is/here/mygulpfile.js') +
-          '\n' +
-        'The current directory : ' + path.join(fixturesDir, 'flags/gulpfile')
+      expect(sliceLines(stdout, 2, 4)).toEqual(
+        'This gulpfile : ' + path.join(baseDir, 'is/here/mygulpfile.js') + '\n' +
+        'The current directory : ' + baseDir
       );
       done(err);
     }
   });
 
-  it('Should configure with a .gulp.* file in the directory specified by ' +
-  '\n\t--cwd', function(done) {
-    runner
-      .gulp('--cwd ./flags/gulpfile')
-      .run(cb);
+  it('Should configure with a .gulp.* file in the directory specified by --cwd', function(done) {
+    exec([
+      'cd ' + path.join(__dirname, 'fixtures/config') + cmdSep,
+      gulpCmd,
+      '--cwd ./flags/gulpfile',
+    ].join(' '), cb);
 
     function cb(err, stdout, stderr) {
-      expect(err).toEqual(null);
+      expect(err).toBeNull();
       expect(stderr).toEqual('');
-      stdout = headLines(stdout, 2, 3);
-      expect(stdout).toEqual(
-        'This gulpfile : ' +
-          path.join(fixturesDir, 'flags/gulpfile/is/here/mygulpfile.js') +
-          '\n' +
-        'The current directory : ' + path.join(fixturesDir, 'flags/gulpfile')
+      expect(sliceLines(stdout, 3, 5)).toEqual(
+        'This gulpfile : ' + path.join(baseDir, 'is/here/mygulpfile.js') + '\n' +
+        'The current directory : ' + baseDir
       );
       done(err);
     }
   });
 
-  it('Should ignore a ./gulp.* file if another directory is specified by ' +
-  '\n\t--cwd', function(done) {
-    runner
-      .chdir('./flags/gulpfile')
-      .gulp('--cwd ./cwd')
-      .run(cb);
+  it('Should load a ./gulp.* file in a directory specified by --cwd', function(done) {
```

Comment:
This test case runs gulp at `test/fixutres/config` and takes options `--cwd ./flags/gulpfile'`.

In `test/fixutres/config` there is no gulpfile and no `.gulp.*`, and in `test/fixtures/config/flags/gulpfile` there is only a `.gulp.json`, so gulp changes CWD to `test/fixtures/config/flags/gulpfile` and loads the `.gulp.json`.

And, since this `.gulp.json` has `flag.gulpfile: "./is/here/mygulpfile.js"`, gulp uses this gulpfile and runs it.

Reply:
This behavior change was due to `.configFiles.project.findUp: true`, but I misunderstood it was due to liftoff’s change.

I’ll modify the directory and file composition under `fixtures/config/flags/gulpfile’ not to mix behaviors due to a config flag and behaviors due to finding up  a config file. In addition, I'll add test cases for find up a config file.